.python-version

```diff
@@ -0,0 +1 @@
+3.10
```

Comment:
I want to pin the Python version and the dependencies for development. This is how I usually do it, but I'm certainly not a pro with this, so if there's a better way let me know :slightly_smiling_face:

Reply:
All good, that how I usually do it too.

Reply:
I usually let poetry decide for itself based on `pyproject.toml`. Right now, it says `3.8` in `pyproject.toml` and `3.10` in `.python-version`. Which one is it? :)

In turn, I check in a `poetry.toml` file with:

```toml
[virtualenvs]
create = true
in-project = true
```

With that, poetry created a local `.venv` folder (instead of in your user directory).

What do you guys think of that approach?

Reply:
> What do you guys think of that approach?

Wow, I didn't know that existed, that's well hidden in the documentation! Fine by me 👍

> Which one is it?

Both! 😊 The `pyproject.toml` specifies _`^3.8`_, which includes `3.10`. I'd like to have reproducible builds of the documentation (and formatting/linting), that's why I'd like to pin the development dependencies (in `poetry.lock`) and the Python version somehow. This way the GitHub workflow to build the docs can use the exact same ones as we do locally.

I'm not set on this though, do you think there are benefits to not locking the dependencies for development?

Reply:
That makes sense. :) It works for me.